### ISSUE TITLE
Instructor Web UI & API Endpoint for Adding a User

### ISSUE THREAD

**Issue by zjxiaohan** [Submitty/Submitty#4168]
**What problem are you trying to solve with Submitty**
Similar to https://github.com/Submitty/Submitty/issues/3398. Let's add a superuser accessible GUI & API endpoint for adding a submitty user.

**Describe the way you'd like to solve this problem**
Adding a submitty user has little to do with linux user groups, just some database commands (as far as I can see from `adduser.py`), so I am in favor of rewriting the logic in PHP and make it accessible from web & API.

**Comment by zjxiaohan** [comment]
Not sure if this is needed or of high priority as there seems to be auto feed. Is this already a well automated process?

**Comment by bmcutler** [comment]
Yes, at RPI we have a custom tool that requires direct access to the database.  Once this is in place we will likely refactor the script to use the API. 

Adding a student to a course via the API should be available to instructor(s) of that course.
But adding users of higher level access, or adding to other courses is only by superuser.

**Comment by MasterOdin** [comment]
`adduser.py` was written for someone who was using DatabaseAuthentication only and looking at it, I never got back to it to put in the stuff necessary for basic PamAuthentication, which would require that a user account gets created on the local machine.

Additionally, at RPI, I believe that we use a more complicated PAM setup, but that I believe at least for faculty and superuser, it may be expected that a local account be created on the machine to allow them to ssh into their courses and such.

**Comment by zjxiaohan** [comment]
I am reading `UsersController`.

Is this piece of code working if we are using PamAuthentication? From @MasterOdin 's info database changes are far from enough when adding a new submitty user. Am I understanding it right?

https://github.com/Submitty/Submitty/blob/2d8d5badc33dd0bec743adf7c7d5935bd6a13233/site/app/controllers/admin/UsersController.php#L339-L344
